async-aggregators/src/utils/q_custom_slider.py

```diff
@@ -0,0 +1,133 @@
+import typing
+
+from PyQt6.QtCore import QPoint, QRect, Qt
+from PyQt6.QtGui import QMouseEvent, QPainter
+from PyQt6.QtWidgets import (
+    QHBoxLayout, QSlider, QStyle, QStyleOptionSlider, QVBoxLayout, QWidget
+)
+
+
+class QCustomSlider(QWidget):
+    """A custom slider with tick labels"""
+
+    def __init__(self, minimum: int, maximum: int, interval: int = 1,
+                 orientation: Qt.Orientation = Qt.Orientation.Horizontal,
+                 labels: list | None = None, p0: typing.Any = 0, parent: typing.Any = None,
+                 suppress_mouse_move: bool = False) -> None:
```

Comment:
Personally I think this is easier to read, but opinions vary.
```suggestion
    def __init__(self,
        minimum: int, maximum: int,
        interval: int = 1,
        orientation: Qt.Orientation = Qt.Orientation.Horizontal,
        labels: list | None = None,
        p0: typing.Any = 0,
        parent: typing.Any = None,
        suppress_mouse_move: bool = False,
    ) -> None:
```